### ISSUE TITLE
Allow clients to disable media loading

### ISSUE THREAD

**Issue by kneekoo** [minetest/minetest#13781]
### Problem

When games don't have enough RAM, Minetest crashes with OOM.

![1i4z5zyi7fya1](https://github.com/minetest/minetest/assets/566306/0350680c-aade-4996-836c-19ca6c1d5b32)

### Solutions

Toggles/checkboxes for media loading in the main menu would be useful, not only, but also to save RAM.

These options would help separate the soundtrack from in-game sounds:
- music
- sounds

Controlling the RAM usage on a device can be very limited in some case - lack of administrative rights, or a mobile device with insufficient headroom for apps requiring a larger chunk of RAM (common for low-mid range mobile devices). That's why being able to prevent RAM usage can save the day.

### Alternatives

Not sure if this can be achieved via game settings. If not, it would be a nice alternative.

### Additional context

Minetest 5.8.0 will improve this situation, but not loading a soundtrack at all will obviously require less RAM.

It would also be nice for the client not to download/cache the media that the user chooses to disable.

**Comment by kneekoo** [comment]
This is not a duplicat because it's a feature request, not a bug report. Saving RAM by disabling media loading can make games playable on devices with low-medium RAM.

> This is a bug in LuaJIT, it's not related to media loading. Please make sure you're using an up-to-date version of LuaJIT with GC64, note that LuaJIT is rolling release now

Off-topic, considering the point of the issue, but this leaves a few questions for those installing Minetest from the website or Google Play:

1. How can an Android user use an up-to-date LuaJIT?
2. With [README.md](https://github.com/minetest/minetest/blob/master/README.md), [main website page](https://www.minetest.net/), and [downloads page](https://www.minetest.net/downloads/) not mentioning LuaJIT, how can one know about it, install and use it with Minetest?

**Comment by erlehmann** [comment]
@rubenwardy I believe that if this is not a duplicate and is actually related to media loading, the context is here: https://git.minetest.land/MineClone2/MineClone2/issues/3874

TL;DR: Newer versionrs of MineClone 2 include background music and people believe that this makes it impossible to load the game on their old Android phones.

**Comment by rubenwardy** [comment]
I'd suggest changing this issue to be specifically about adding support for music:

- A new type of media
- Can be muted/volume changed independently from sounds (if muted, the media shouldn't be downloaded)
- Should be streamed from the server (but doesn't need to be in the first version)

**Comment by kneekoo** [comment]
As the title says, as well as the opening issue, this is important when you want to save RAM. Of course it's also useful for people who don't want to hear any music in the menu and in-game.

Streaming the music would be ideal to reduce the amount of used RAM. In this case, it might be possible even for some people with low RAM to listen to the soundtrack. But I think it would be best to stream it from a local device (cache), to save bandwidth on the server and data traffic for the clients.

**Comment by Desour** [comment]
FYI, mt 5.8-dev already doesn't decompress sound files until they are played. Mt 5.7 did decompress everything at connect.
Compressed ogg vorbis music is usually <10MiB per piece, this should easily fit into ram. PCM data on the other hand is about 0.18MiB (`44.1k*2B*2`) per second (10MiB per minute).
Implementing unloading of PCM data is on my TODO list.

(Theoretically, we could also make use of the `mute_sound` setting, by not loading sounds if it's enabled. But we'd have to keep track of the time passed since mute and do other stuff. I'm not planning to do this right now. If someone really doesn't want any sounds, they can use `enable_sound`.)

Anyway, if my calculations are not completely off, you'd need multiple hours of music to trigger an OOM error just from music. It would make sense to do some sort of profiling to see what else could be done to use less memory.

**Comment by kneekoo** [comment]
It will take some time until 5.8.0 gets released, so adding a soundtrack before that wasn't the ideal choice for MineClone2, but it's great to know only one track gets uncompressed with the next release. It would still be great to avoid the media download/caching when the music is disabled by the users. It saves storage space, wait time, and I/O.

**Comment by rubenwardy** [comment]
> As the title says, as well as the opening issue, this is important when you want to save RAM. Of course it's also useful for people who don't want to hear any music in the menu and in-game.
> 
> Streaming the music would be ideal to reduce the amount of used RAM. In this case, it might be possible even for some people with low RAM to listen to the soundtrack. But I think it would be best to stream it from a local device (cache), to save bandwidth on the server and data traffic for the clients.

Streaming music from the server would save bandwidth and data traffic as otherwise you'd need to send all music to all clients, even if the music is never played - increasing load times. Using dynamic media, a mod can only send music to clients when it is needed. However, dynamic media is never freed so you'd end up running out of ram after a long playing session

**Comment by kneekoo** [comment]
Caching on storage and loading into RAM should be separate items. If the client wants music, it should be cached on storage, played back and released from RAM after completion. I don't see why tracks should be kept in RAM until the session ends. That only makes sense for sound effects - the soundtrack doesn't need instant playback.

Ideally, the music would be streamed from the storage, so that RAM usage becomes virtually a non-issue when dealing with media. But even if only one track is loaded into RAM at a time, then released, it's still pretty good.
